### ISSUE TITLE
Slow editing with big XML (20MB)

### ISSUE THREAD

**Issue by NN---** [notepad-plus-plus/notepad-plus-plus#3442]
Open this file in Notepad++ .
Editing is very very slow.

https://mega.nz/#!G5hXCZrQ!LsCdTBMbNzT8z2lAJLPy4ZdjWUyY-sc1YcoDxTBOszc

**Comment by dail8859** [comment]
Post your debug info,  and make sure to turn off auto- completion.

**Comment by NN---** [comment]
Disabling doesn't help.

Notepad++ v7.4.2   (64-bit)
Build time : Jun 18 2017 - 23:38:43
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS : Windows 10 (64-bit)
Plugins : none

**Comment by dail8859** [comment]
This is due to the excessively long line. Scintilla has known performance issues when lines are very long.

**Comment by NN---** [comment]
Do they know about this issue ?
Maybe some profiling is needed there.

**Comment by dail8859** [comment]
Yes they are aware. There is some discussion [here](https://groups.google.com/forum/#!topic/scintilla-interest/jZI6WjxjT9E), and at least on [this post](https://sourceforge.net/p/scintilla/bugs/1916/#b511) on sourceforge the author of Scintilla states:

> very long lines are slow

**Comment by dail8859** [comment]
Also [this issue](https://sourceforge.net/p/scintilla/bugs/1563/) on sourceforge.
